src/ui/components/auth/SignInContent.tsx

```diff
@@ -23,60 +24,84 @@ interface SignInContentProps {
 	onForgotPasswordBtnClick: () => void;
 }
 
+interface IErrorObject {
+	email: string;
+	password: string;
+}
+
 const SignInContent: React.FC<SignInContentProps> = ({
 	onSuccess,
 	onCreateNewUserClick,
 	onForgotPasswordBtnClick,
 }): JSX.Element => {
 	const [isLoading, setIsLoading] = useState<boolean>(false);
+	const [userTriedToSubmit, setUserTriedToSubmit] = useState(false);
 
 	const [email, setEmail] = useState("");
 	const [password, setPassword] = useState("");
-	const [emailError, setEmailError] = useState("");
-	const [passwordError, setPasswordError] = useState("");
 
-	const validateKey = async () => {
-		setIsLoading(true);
+	const [errors, setErrors] = useState<IErrorObject>({
+		email: "",
+		password: "",
+	});
+	const [serverValidationError, setServerValidationError] = useState("");
+
+	const validateCredentials = async () => {
 		const response = await fetchData({
-			url: getApiUrl("/api/key/validate"),
+			url: getApiUrl("/api/sample/validate"),
 			method: "POST",
 			config: {
 				headers: {
 					// authorization: `Bearer ${apiKey}`,
 				},
 			},
 		});
-
 		const body = await response.json();
-
-		if (response.status === 200 && body.status === "OK") {
-			// localStorageHandler.setItem(StorageKeys.API_KEY, apiKey);
-			onSuccess();
-		} else if (response.status === UNAUTHORISED_STATUS) {
-			// setApiKeyFieldError("Invalid API Key");
+		if (response.status === HTTPStatusCodes.OK) {
+			if (body.status === "OK") onSuccess();
+			else setServerValidationError("Incorrect email and password combination");
 		} else {
-			// setApiKeyFieldError("Something went wrong");
+			setServerValidationError("Something went wrong");
 		}
+	};
 
-		setIsLoading(false);
+	const checkValuesForErrors = () => {
+		const _errors = {
+			email: "",
+			password: "",
+		};
+		if (!email) _errors.email = "Email cannot be empty";
+		if (!password) _errors.password = "Password cannot be empty";
+		if (!validateEmail(email)) _errors.email = "Email is invalid";
+		setErrors(_errors);
+		return Object.values(_errors).some((error) => error);
 	};
 
-	const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
-		e.preventDefault();
-		// setApiKeyFieldError("");
+	const clearErrors = (key: keyof IErrorObject) => {
+		setErrors({ ...errors, [key]: "" });
+		setServerValidationError("");
+	};
 
-		if (!email) {
-			setEmailError("Email cannot be empty");
-		}
-		if (!password) {
-			setPasswordError("Password cannot be empty");
-		}
+	useEffect(() => {
+		if (email && errors.email) clearErrors("email");
+	}, [email]);
+
+	useEffect(() => {
+		if (password && errors.password) clearErrors("password");
+	}, [password]);
 
-		// if (apiKey !== null && apiKey !== undefined && apiKey.length > 0) {
-		// 	void validateKey();
-		// } else {
-		// 	setApiKeyFieldError("API Key field cannot be empty");
-		// }
+	const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+		e.preventDefault();
```

Comment:
It may be a bad idea to do this, if in the future we want to add analytics on a global level then this will prevent that from being called.